routes/notes.route.js

```diff
@@ -0,0 +1,106 @@
+import express from "express";
+import authMiddleware from '.././middlewares/auth.middleware.js';
+import notesService from '.././services/note.service.js';
+import { body,check,oneOf,param,query, validationResult } from 'express-validator';
+
+const notesRouter = express.Router();
+
+notesRouter.get('/api/notes',
+    async (req,res,next)=>{
+        //request non autenticata
+        //Se la query string non contiene filtri 
+        if(!req.query.date && !req.query.limit)
+            res.status(200)
+            .json({'success' : true,
+                'list' : true,
+                'data' : await notesService.getAll()})
+        else
+            next();
+    },
+    authMiddleware,
+    oneOf([query('limit').exists(),query('date').exists()]),
+    async (req,res,next)=>{
+        //request autenticata
+        //Se la query string contiene i filtri date and/or limit
+        try{
+            validationResult(req).throw()
+            res.status(200)
+            .json(await notesService.getFiltered(req.query.date,
+                                        req.query.limit)
+                                        )
+        }catch(err){
+            res.status(400)
+            .json(notesService.badRequest())
+        }
+        
+    }
+);
+
+notesRouter.get('/api/notes/:uuid',param('uuid').isUUID(),async (req,res)=>{
+    try{
+        validationResult(req).throw()
+        let note = await notesService.getOneById(req.params.uuid)
+        if(note)
+            res.status(200)
+            .json({
+                "success": true,
+                "single": true,
+                "data" : note
+            })
+        else
+            res.status(400)
+            .json(notesService.badRequest())
+    }catch(err){
+        res.status(400)
+            .json(notesService.badRequest())
+    }
+    
+})
+
+notesRouter.post('/api/notes',authMiddleware,
+    body('user').isAlphanumeric(),
+    body('date').isDate(),
+    body('title').isString(),
+    body('body').isString(),
+async (req,res)=>{
+    try{
+        validationResult(req).throw()
+        //Se body contiene uuid = bad request
+        if(('uuid' in req.body))
+            res.status(400)
+            .json(notesService.badRequest())
+        //Se body non contiene tutti i dati = bad request
+        if(!(('user') in req.body) ||
```

Comment:
Ricordami di vederlo insieme in classe. Dovresti essere l'unico ad aver fatto ed è una chicca.